metadata_service/proxy/atlas_proxy.py

```diff
@@ -182,11 +179,12 @@ def _create_reader(self, metadata_guid: str, user_guid: str, reader_qn: str, tab
             'attributes': {'qualifiedName': reader_qn,
                            'isFollowing': True,
                            'count': 0,
-                           'entityMetadata': {'guid': metadata_guid},
-                           'user': {'guid': user_guid},
-                           'entityUri': table_uri}
+                           'user': {'guid': user_guid}}
         }
-        self._driver.entity_bulk.create(data={'entities': [reader_entity]})
+
+        readers = table_entity.entity[self.ATTRS_KEY]['readers'].append(reader_entity)
```

Comment:
Are you making sure the uniqueness of relation between a user and a table reader ?

Reply:
this should be handled by atlas if reader entity is created first.